hw/ip/otbn/dv/otbnsim/sim/state.py

```diff
@@ -151,9 +201,24 @@ def rnd_completed(self) -> None:
         self.rnd_cdc_pending = False
         self.rnd_cdc_counter = 0
 
+    def urnd_completed(self) -> None:
+        # URND completed gets called after RTL signals that the processing
+        # of incoming EDN data is done. This also sets up EXEC state of the
+        # FSM of the model.
+        assert self.urnd_cdc_pending
+        assert self.urnd_cdc_counter < 6
+
+        assert self.fsm_state == FsmState.PRE_EXEC
```

Comment:
How do you ensure this in system tests? Before, there was a hack in `stepped.py` but I think you're getting rid of that. Don't things explode?

Reply:
I tried to add a similar hack by only doing the EDN processing when we finish `urnd_step`s in the model itself. This will be properly done after we implemented EDN requests from model side.